src/WebJobs.Extensions.OpenAI/OpenAIWebJobsBuilderExtensions.cs

```diff
@@ -60,7 +61,14 @@ public static IWebJobsBuilder AddOpenAIBindings(this IWebJobsBuilder builder)
         // Service objects that will be used by the extension
         builder.Services.AddSingleton<TextCompletionConverter>();
         builder.Services.AddSingleton<EmbeddingsConverter>();
-        builder.Services.AddSingleton<SemanticSearchConverter>();
+        
+        SemanticSearchConverter semanticSearchConverter = null!;
+        builder.Services.AddSingleton(serviceProvider =>
```

Comment:
Why not just do `builder.Services.AddSingleton<SemanticSearchConverter>`? I think that would be equivalent to what you're doing below, but less typing.

Reply:
updated